gaia-fast-list.js

```diff
@@ -1111,9 +1154,21 @@ Internal.prototype = {
   updateCachedHtml() {
     if (!this.el.caching) return;
     debug('update cached html');
-    var items = this.el.querySelectorAll('.gfl-item, .gfl-section');
-    var html = [].map.call(items, el => el.outerHTML).join('');
-    this.setCache('html', html);
+    var maxViewportHeight = Math.max(window.innerWidth, window.innerHeight);
+    var length = Math.ceil(maxViewportHeight / this.getItemHeight());
+    var items = [].slice.call(this.el.querySelectorAll('.gfl-item'), 0, length);
+
+    // remove any attributes to save bytes
+    var itemsHtml = items.map(el => el.outerHTML)
+      .join('')
+      .replace(/style\=\"[^"]+\"/g, '')
+      .replace(/data-tweak-delta=""/g, '');
+
+    var sections = [].slice.call(this.el.querySelectorAll('.gfl-section'));
```

Comment:
... and thus we can also limit the number of sections stored. We will need to measure the sections to ensure that we are caching enough to fill the viewport.